diff --git a/node_modules/trezor-link/lib/bridge/v2.js b/node_modules/trezor-link/lib/bridge/v2.js
index 7acedea..1ba7306 100644
--- a/node_modules/trezor-link/lib/bridge/v2.js
+++ b/node_modules/trezor-link/lib/bridge/v2.js
@@ -429,6 +429,25 @@ var BridgeTransport = (_class = (_temp = /*#__PURE__*/function () {
 
               case 9:
                 jsonData = (0, _receive.receiveOne)(messages, new Buffer(resData, "hex"));
+                if (name === 'GetFeatures' || name === 'Initialize') {
+                    // 某些固件的 bug，major_version 会为 0，默认为 1
+                    if (jsonData.message.major_version < 1) {
+                        jsonData.message.major_version = 1;
+                    }
+
+                    // onekey 固件为 2.0.0 时，需要将 major_version 改为 1 防止逻辑错误
+                    // 同时使用 onekey_version 字段标记正确的版本，将原来字段版本改为 1.9.8 是 onekey 支持的 trezor 固件的最后一个大版本
+                    if (
+                        jsonData.message.major_version === 2
+                        && jsonData.message.minor_version === 0
+                        && jsonData.message.patch_version === 0
+                    ) {
+                        jsonData.message.onekey_version = '2.0.0';
+                        jsonData.message.major_version = 1;
+                        jsonData.message.minor_version = 9;
+                        jsonData.message.patch_version = 8;
+                    }
+                }
                 return _context10.abrupt("return", check.call(jsonData));
 
               case 11:
